### ISSUE TITLE
Default-button focus ring has wrong width in Workflow mode shown during Setup Assistant / login window

### ISSUE THREAD

**Issue by villasenor** [swiftDialog/swiftDialog#671]
**Describe the bug**
The focus ring for buttons does not always match the actual size of the button, especially in Card/Workflow mode when using custom button text. This bug was visible even before custom next and previous button text was added if you used custom text for the final card's finish button. I only really notice this when Dialog is shown during Setup Assistant as an overlay (which works great by the way!). I haven't tested but this is probably reproducible at the login window as well.

**To Reproduce**
1. Launch Workflow mode Dialog during Setup Assistant from a LaunchDaemon
1. Use Workflow mode with long and short text in buttons
2. Navigate previous and next

**Expected behavior**
Either no focus ring, or the focus ring matches the size of the button

**Screenshots**

<img width="821" height="600" alt="Image" src="https://github.com/user-attachments/assets/bf29d9ac-c962-4974-b0be-58a4e17ed0fc" />

**Desktop:**
 - macOS version 26.4.1
 - swiftDialog version 3.1.0 Beta 3

**Additional context**
Doing some light research, it seems like it could be a view-identity timing thing or ButtonBarView not getting .id(currentCardIndex) recreation on card transitions? NSButton being parented to a previous frame? Still learning SwiftUI so not sure. Happy to test different solutions! :)

**Comment by bartreardon** [comment]
This is a swiftui rendering bug and is present in other apps
As an example, Apples Developer app

Feedbacks have been raised

<img width="819" height="631" alt="Image" src="https://github.com/user-attachments/assets/1eeafe48-6af9-4b8c-aa19-f6143f8df78c" />

**Comment by villasenor** [comment]
Woah! Thanks for submitting Feedback to Apple. Do you think there is any way it can be worked around while Apple looks into it?

**Comment by bartreardon** [comment]
There is already code in SD to try and mitigate against it but it doesn't always work. It's a redraw issue where the bounds of the view changes after the highlight has been rendered. Not a lot else we can do I'm afraid. Apple's bug to fix.

**Comment by villasenor** [comment]
Makes sense, thanks for the explanation!
